wai-extra/Network/Wai/Middleware/RequestLogger/JSON.hs

```diff
@@ -73,19 +76,26 @@ word32ToHostAddress = T.intercalate "." . map (T.pack . show) . fromIPv4 . fromH
 readAsDouble :: String -> Double
 readAsDouble = read
 
-requestToJSON :: NominalDiffTime -> Request -> [S8.ByteString] -> Value
-requestToJSON duration req reqBody =
-  object
+-- | Get the JSON output for a request
+--
+-- This gives a sensible JSON representation for the given request. You can
+-- optionally include a request body, as well as a duration.
+--
+-- @since 3.1.4
+requestToJSON ::  Request -> [S8.ByteString] -> Maybe NominalDiffTime -> Value
```

Comment:
Some Haddocks on these parameters to explain what these parameters are for would be helpful